### DISCUSSION TITLE
strange `<slash>` example in MusicXML standard?

### DISCUSSION THREAD

**Discussion by lemzwerg** [w3c-cg/musicxml#546] [General]
Looking at https://www.w3.org/2021/06/musicxml40/musicxml-reference/examples/slash-element/ I wonder what's going on.  How comes that there are four beats shown in the image for a single, whole note in the MusicXML code?  Is this really correct?

**Comment by mscuthbert** [comment]
This is, I believe, correct.  The `<slash>` element specifies at what point the notation will use slashes at regular intervals rather than displaying notation.  I think it would be better if the default reference example showed the optional `<slash-type>quarter</slash-type>` since it would make it less mysterious what is happening.  I will make an issue to do that.

**Reply by lemzwerg** [reply]
OK, thanks.  It certainly helps to use `<slash-type>` in the example, but please also add some words to the description of `<slash>` to point out that a note's `<duration>` value gets ignored while `<slash>` is active.

**Reply by mscuthbert** [reply]
It is only ignored for display.  It can be quite important for playback.

**Reply by lemzwerg** [reply]
Yes, I meant 'ignored' just for the typesetting case.  Sorry for being imprecise.

Anyway, I think that still some documentation is missing to avoid undefined spots in the specification.  In particular, I wonder what's happening if the note lengths are not integer multiples of the 'beat', and how rests are treated in such situations.  Consider this example (using slashes without stems).

![image](https://github.com/user-attachments/assets/10a7cee6-6811-469b-a9fd-7e8ae681813e)

I'm not a jazz guy, but the rest in the staff with the slashes looks wrong.  If my assumption is correct, there should be some clarifications about the granularity.  Maybe something like the following?

"To make the display of stemless slashes work correctly, note lengths must be integer multiples of the slash beat."

**Reply by mdgood** [reply]
No, this is wrong. In 4/4 with stemless slash notation, you have four slashes. It doesn't matter what the underlying music is. That's the point of slash notation.

**Reply by lemzwerg** [reply]
> No, this is wrong.

Thanks for clarification.  I expected exactly that.  However, what about rests in general like the following?

![image](https://github.com/user-attachments/assets/1579cbc6-b183-4cf3-97f9-0cb5572cbeb1)

Again, I'm not a jazz guy and I don't know whether this is valid.

**Reply by mdgood** [reply]
If you see something like this, you would use `<slash type="stop"/>` before the rest(s) and `<slash type="start"/>` after the rest. You wouldn't generally see the quarter rest in that context, though.

**Reply by lemzwerg** [reply]
Ah, ok, thanks.  How should I know :slightly_smiling_face: 

Seriously, I think this is not obvious and should be mentioned in the docs.

**Reply by mdgood** [reply]
I think the [`<except-voice>` example](https://www.w3.org/2021/06/musicxml40/musicxml-reference/examples/except-voice-element/) clearly shows slashes being used where the underlying music is rests.

**Reply by lemzwerg** [reply]
It does, yes, but IMHO it would be helpful to explicitly mention what you have written above about the 'stemless slash mode', namely that "It doesn't matter what the underlying music is."

**Reply by mscuthbert** [reply]
I think that in the decades of the specification, we have never heard from jazz users that the specification is unclear.  I know that it could be difficult for people who don't play from slash notation, but the same is true for all music notation for people who don't regularly read it.

**Reply by lemzwerg** [reply]
Well, now you hear that the specification is unclear :slightly_smiling_face: 

My point of view is that it doesn't hurt if the specification is not as terse as possible but a bit verbose.  Obviously, you think differently, but please bear in mind that only a small number of people who are going to implement the MusicXML specification is well-versed in all music styles.  What is obvious for me isn't obvious for you and vice versa.

**Reply by lemzwerg** [reply]
A last issue (at least for now). I wrote:

>  I wonder what's happening if the note lengths are not integer multiples of the 'beat',

This is still an open question, that is, what happens if a stemless slash section starts or ends off-beat?  I know that this belongs into the category 'don't do that', but users probably have the weirdest ideas how to (ab)use a standard, and I like to see border cases handled (even if simply tagged as 'this is undefined behaviour').
